### ISSUE TITLE
[Bug] The built-in logging for python 3.6 doesn't have a "stacklevel" keyword arguments. Leads to error when using CachedDataset.

### ISSUE THREAD

**Issue by Allan-Yoriichi** [neuromorphs/tonic#189]
## traceback info
```commandline
Traceback (most recent call last):
  File "/home/allan/PycharmProjects/snn_classifier_trainer/examples/nmnist_example.py", line 91, in <module>
    for inp_ts, tgt_ts in train_loader:
  File "/home/allan/PycharmProjects/snn_classifier_trainer/venv/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 517, in __next__
    data = self._next_data()
  File "/home/allan/PycharmProjects/snn_classifier_trainer/venv/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 557, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/allan/PycharmProjects/snn_classifier_trainer/venv/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/allan/PycharmProjects/snn_classifier_trainer/venv/lib/python3.6/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/allan/PycharmProjects/snn_classifier_trainer/venv/lib/python3.6/site-packages/tonic/cached_dataset.py", line 124, in __getitem__
    stacklevel=2,
  File "/usr/lib/python3.6/logging/__init__.py", line 1902, in info
    root.info(msg, *args, **kwargs)
  File "/usr/lib/python3.6/logging/__init__.py", line 1308, in info
    self._log(INFO, msg, args, **kwargs)
TypeError: _log() got an unexpected keyword argument 'stacklevel'
```

## reason
which is caused by the code in `__getitem__` method of `DiskCachedDataset` below the ` except (FileNotFoundError, OSError) as _` statement:

```python
    def __getitem__(self, item) -> Tuple[object, object]:
        copy = np.random.randint(self.num_copies)
        file_path = os.path.join(self.cache_path, f"{item}_{copy}.hdf5")
        try:
            data, targets = load_from_disk_cache(file_path)
        except (FileNotFoundError, OSError) as _:
            logging.info(
                f"Data {item}: {file_path} not in cache, generating it now",
                stacklevel=2,
            )
            data, targets = self.dataset[item]
            save_to_disk_cache(
                data, targets, file_path=file_path, compression=self.compression
            )
            # format might change during save to hdf5,
            # i.e. tensors -> np arrays
            data, targets = load_from_disk_cache(file_path)

        if self.transform:
            data = self.transform(data)
        if self.target_transform:
            targets = self.target_transform(targets)
        return data, targets
```
So I need to changge my interpreter to python3.8 or is there any convient way for this? Many thanks!

----update----
By directly remove this `stacklevel` argument, this error could be solved.

**Comment by biphasic** [comment]
Hey Yalun,
sorry that you're encountering the error. Python 3.6 is EOL https://endoflife.date/python so that's why it's not included in the test pipeline, however I'd accept a PR that fixes your issue if you like. Thank you!

**Comment by biphasic** [comment]
won't actively develop fixes for Python 3.6 so closing for now.
